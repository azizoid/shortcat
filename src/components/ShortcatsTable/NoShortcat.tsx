import { Table } from "flowbite-react";

export const NoShortcat = () => <Table.Row className="bg-white dark:border-gray-700 dark:bg-gray-800" key="not-found">
  <Table.Cell className="whitespace-nowrap font-medium text-center text-gray-900 dark:text-white" colSpan={5}>
    Your list is empty Click
  </Table.Cell>
</Table.Row>