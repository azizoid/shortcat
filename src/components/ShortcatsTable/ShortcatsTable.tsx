import { Shortcat } from "@prisma/client";
import { Table } from "flowbite-react";
import Link from "next/link";

type ShortcatsTableProps = {
  data: Shortcat[]
}
export const ShortcatsTable = ({ data }: ShortcatsTableProps) => <Table hoverable={true}>
  <Table.Head>
    <Table.HeadCell>
      GUID
    </Table.HeadCell>
    <Table.HeadCell>
      Redirect Url
    </Table.HeadCell>
    <Table.HeadCell>
      Status
    </Table.HeadCell>
    <Table.HeadCell>
      <span className="sr-only">
        View
      </span>
    </Table.HeadCell>
    <Table.HeadCell>
      <span className="sr-only">
        Edit
      </span>
    </Table.HeadCell>
  </Table.Head>
  <Table.Body className="divide-y">
    {data?.length === 0 && <Table.Row className="bg-white dark:border-gray-700 dark:bg-gray-800" key="not-found">
      <Table.Cell className="whitespace-nowrap font-medium text-center text-gray-900 dark:text-white" colSpan={5}>
        Your list is empty Click
      </Table.Cell>
    </Table.Row>}

    {data?.map(shortcat =>
      <Table.Row className="bg-white dark:border-gray-700 dark:bg-gray-800" key={shortcat.id}>
        <Table.Cell className="whitespace-nowrap font-medium text-gray-900 dark:text-white">
          {shortcat.shortcode_guid}
        </Table.Cell>
        <Table.Cell>
          {shortcat.redirect_url}
        </Table.Cell>
        <Table.Cell>
          {shortcat.active ? 'Active' : 'Inactive'}
        </Table.Cell>
        <Table.Cell>
          <Link href={`/${shortcat.shortcode_guid}`} >
            View
          </Link>
        </Table.Cell>
        <Table.Cell>
          <Link href={`/${shortcat.shortcode_guid}/edit`} >
            Edit
          </Link>
        </Table.Cell>
      </Table.Row>)}
  </Table.Body>
</Table>