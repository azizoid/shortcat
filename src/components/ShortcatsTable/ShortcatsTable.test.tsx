import { render } from '@testing-library/react'
import { Shortcat } from '@prisma/client'
import { ShortcatsTable } from './ShortcatsTable'

const mockShortcats: Shortcat[] = [
  {
    id: 1,
    shortcode_guid: 'ABC123',
    redirect_url: 'https://mojkuran.com',
    active: true,
  },
  {
    id: 2,
    shortcode_guid: 'DEF456',
    redirect_url: 'https://quran.az',
    active: false,
  },
]

describe('ShortcatsTable', () => {
  it('should render a table with the correct headings', () => {
    const { container, getByText } = render(<ShortcatsTable data={[]} />)

    expect(getByText('GUID')).toBeInTheDocument()
    expect(getByText('Redirect Url')).toBeInTheDocument()
    expect(getByText('Status')).toBeInTheDocument()
    expect(getByText('View')).toBeInTheDocument()
    expect(getByText('Edit')).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })

  it('should render a message when there is no data', () => {
    const { getByText } = render(<ShortcatsTable data={[]} />)

    expect(getByText('Your list is empty. Click the button on the top to add a Shortcat.')).toBeInTheDocument()
  })

  it('should render a table with the provided data', () => {
    const { container, getByText } = render(<ShortcatsTable data={mockShortcats} />)

    expect(getByText('ABC123')).toBeInTheDocument()
    expect(getByText('https://mojkuran.com')).toBeInTheDocument()
    expect(getByText('Active')).toBeInTheDocument()

    expect(getByText('DEF456')).toBeInTheDocument()
    expect(getByText('https://quran.az')).toBeInTheDocument()
    expect(getByText('Inactive')).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
})