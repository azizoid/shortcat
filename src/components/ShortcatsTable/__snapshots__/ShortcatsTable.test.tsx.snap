// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortcatsTable should render a table with the correct headings 1`] = `
<div>
  <div
    class="relative overflow-x-auto shadow-md sm:rounded-lg"
    data-testid="table-element"
  >
    <table
      class="w-full text-left text-sm text-gray-500 dark:text-gray-400"
    >
      <thead
        class="bg-gray-50 text-xs uppercase text-gray-700 dark:bg-gray-700 dark:text-gray-400"
      >
        <tr>
          <th
            class="px-6 py-3"
          >
            GUID
          </th>
          <th
            class="px-6 py-3"
          >
            Redirect Url
          </th>
          <th
            class="px-6 py-3"
          >
            Status
          </th>
          <th
            class="px-6 py-3"
          >
            <span
              class="sr-only"
            >
              View
            </span>
          </th>
          <th
            class="px-6 py-3"
          >
            <span
              class="sr-only"
            >
              Edit
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        class="divide-y"
      >
        <tr
          class="hover:bg-gray-50 dark:hover:bg-gray-600 bg-white dark:border-gray-700 dark:bg-gray-800"
          data-testid="table-row-element"
        >
          <td
            class="px-6 py-4 whitespace-nowrap font-medium text-center text-gray-900 dark:text-white"
            colspan="5"
          >
            Your list is empty. Click the button on the top to add a Shortcat.
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`ShortcatsTable should render a table with the provided data 1`] = `
<div>
  <div
    class="relative overflow-x-auto shadow-md sm:rounded-lg"
    data-testid="table-element"
  >
    <table
      class="w-full text-left text-sm text-gray-500 dark:text-gray-400"
    >
      <thead
        class="bg-gray-50 text-xs uppercase text-gray-700 dark:bg-gray-700 dark:text-gray-400"
      >
        <tr>
          <th
            class="px-6 py-3"
          >
            GUID
          </th>
          <th
            class="px-6 py-3"
          >
            Redirect Url
          </th>
          <th
            class="px-6 py-3"
          >
            Status
          </th>
          <th
            class="px-6 py-3"
          >
            <span
              class="sr-only"
            >
              View
            </span>
          </th>
          <th
            class="px-6 py-3"
          >
            <span
              class="sr-only"
            >
              Edit
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        class="divide-y"
      >
        <tr
          class="hover:bg-gray-50 dark:hover:bg-gray-600 dark:border-gray-700 dark:bg-gray-800 bg-white"
          data-testid="table-row-element"
        >
          <td
            class="px-6 py-4 whitespace-nowrap font-medium text-gray-900 dark:text-white"
          >
            ABC123
          </td>
          <td
            class="px-6 py-4"
          >
            https://mojkuran.com
          </td>
          <td
            class="px-6 py-4"
          >
            Active
          </td>
          <td
            class="px-6 py-4"
          >
            <a
              href="/ABC123"
            >
              View
            </a>
          </td>
          <td
            class="px-6 py-4"
          >
            <a
              href="/ABC123/edit"
            >
              Edit
            </a>
          </td>
        </tr>
        <tr
          class="hover:bg-gray-50 dark:hover:bg-gray-600 dark:border-gray-700 dark:bg-gray-800 bg-gray-50"
          data-testid="table-row-element"
        >
          <td
            class="px-6 py-4 whitespace-nowrap font-medium text-gray-900 dark:text-white"
          >
            DEF456
          </td>
          <td
            class="px-6 py-4"
          >
            https://quran.az
          </td>
          <td
            class="px-6 py-4"
          >
            Inactive
          </td>
          <td
            class="px-6 py-4"
          >
            <a
              href="/DEF456"
            >
              View
            </a>
          </td>
          <td
            class="px-6 py-4"
          >
            <a
              href="/DEF456/edit"
            >
              Edit
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
